src/main/java/core/basesyntax/figure/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.figure;
+
+import core.basesyntax.figuretype.Circle;
+import core.basesyntax.figuretype.IsoscelesTrapezoid;
+import core.basesyntax.figuretype.Rectangle;
+import core.basesyntax.figuretype.RightTriangle;
+import core.basesyntax.figuretype.Square;
+import core.basesyntax.misc.ColorSupplier;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int UNITS = 100;
+    private Random random = new Random();
+    private ColorSupplier colorSupply = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int r1 = random.nextInt(UNITS);
+        int r2 = random.nextInt(UNITS);
+        int r3 = random.nextInt(UNITS);
```

Comment:
let's generate random value only if we need it
in case when we will have Decagon will we create ten new variables with a random values?